frontend/src/pages/modelServing/__tests__/useModelServingProjects.spec.ts

```diff
@@ -0,0 +1,78 @@
+import { k8sListResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { act } from '@testing-library/react';
+import { ProjectKind } from '~/k8sTypes';
+import { standardUseFetchState, testHook } from '~/__tests__/unit/testUtils/hooks';
+import useModelServingProjects from '~/pages/modelServing/useModelServingProjects';
+
+jest.mock('@openshift/dynamic-plugin-sdk-utils', () => ({
+  k8sListResource: jest.fn(),
+}));
+
+const k8sListResourceMock = k8sListResource as jest.Mock;
+
+describe('useModelServingProjects', () => {
+  it('should return model serving projects when namespace is not provided', async () => {
+    const mockModelServingProjects = {
+      items: [{ metadata: { name: 'item 1' } }, { metadata: { name: 'item 2' } }] as ProjectKind[],
+    };
+    k8sListResourceMock.mockReturnValue(Promise.resolve(mockModelServingProjects));
+
+    const renderResult = testHook(useModelServingProjects)();
+    expect(k8sListResourceMock).toHaveBeenCalledTimes(1);
```

Comment:
Same comment as https://github.com/opendatahub-io/odh-dashboard/pull/2375/files#r1463980771 to add more robust tests.